### ISSUE TITLE
Pointer file error on traced empty file

### ISSUE THREAD

**Issue by tbhartman** [git-lfs/git-lfs#1920]
I am running:

```
git-lfs/1.5.5 (GitHub; windows amd64; go 1.7.4; git c2dcd6f5)
git version 2.10.0.windows.1
```

summary of how I hit this
----

There are thousands of files in my repo that may grow to several megabytes each, so I just track them all with LFS so that my repo stays a manageable size.  Hoewever, some of the files with the same extension are currently zero bytes, though that might change in the future.

If I try to track a zero byte file with LFS, LFS does not track it.  The commit is OK, but when I go to checkout that commit, I get a `Pointer file error`.

steps to reproduce
----

```sh
# create repo
git init
touch .gitattributes
git add .gitattributes
git commit -m "initial commit"

# create empty file, track, and commit
touch empty_file
git lfs install --local
git lfs track empty_file
git add empty_file .gitattributes
git commit -m "commit empty lfs file"

# switch between commits
git checkout HEAD~1
git checkout master
```

When I checkout `master`, I get this error:

```
Pointer file error: Unable to parse pointer at: "empty_file"
Previous HEAD position was ec8c116... initial commit
Switched to branch 'master'
```

**Comment by technoweenie** [comment]
Thanks for the report. Does this prevent you from working in the repository? Is your `git status` clean, or does it show changes for the empty files?

I tried this on `master` (which will eventually be LFS v2.0), and so no errors. Just a (different) warning:

```sh
$ git checkout master
Previous HEAD position was 4ae1cda... Create README.md
Switched to branch 'master'
Your branch is ahead of 'origin/master' by 2 commits.
  (use "git push" to publish your local commits)
Encountered 2 file(s) that should have been pointers, but weren't:
	empty.txt
	empty_file
```

The problem here is that Git LFS doesn't actually replace empty files with an LFS pointer, so the parser is confused that there's no data to read. Should be easy to handle empty files, since we know what the lfs oid of an empty string will be.

**Comment by tbhartman** [comment]
The files are correct after checkout, I was just confused.  I thought LFS was doing something wrong because these files didn't show up in `git lfs status`.  I then thought *I* was doing something wrong because I got that `Pointer file error` for each file when I went to checkout.

It took me a while to figure out the files that were causing the problem were all empty.

I am new to LFS and am still trying to figure out the flow of it.  This issue just threw me for a loop.
